clang/lib/CodeGen/CGBuiltin.cpp

```diff
@@ -5908,7 +5908,7 @@ RValue CodeGenFunction::EmitBuiltinExpr(const GlobalDecl GD, unsigned BuiltinID,
           }
         }
 
-        assert(PTy->canLosslesslyBitCastTo(FTy->getParamType(i)) &&
+        assert(ArgValue->getType()->canLosslesslyBitCastTo(PTy) &&
```

Comment:
Is there much point to this assertion, given that the BitCast constructor will also assert castIsValid? canLosslesslyBitCastTo looks arbitrarily different from castIsValid

Reply:
`canLosslesslyBitCastTo` is supposed to be stricter than `castIsValid` but it looks too conservative. For example, it doesn't allow bitcast between `float` and `int`.